pkg/resourcecreator/networkpolicy.go

```diff
@@ -2,87 +2,127 @@ package resourcecreator
-		networkPolicyPeer := networking.NetworkPolicyPeer{
+
+func ingressPolicy(app *nais.Application) *[]networkingv1.NetworkPolicyIngressRule {
+	if app.Spec.AccessPolicy.Ingress.AllowAll {
+		return &[]networkingv1.NetworkPolicyIngressRule{{}} // æh funker dette?
+	}
+
+	if len(app.Spec.AccessPolicy.Ingress.Rules) > 0 {
+		policies := networkingv1.NetworkPolicyIngressRule{
+			From: addNetworkPolicyIngressRules(app),
+		}
+		return &[]networkingv1.NetworkPolicyIngressRule{policies}
+	}
+
+	return &[]networkingv1.NetworkPolicyIngressRule{}
+}
+
+
+func addNetworkPolicyEgressRules(app *nais.Application) (networkPolicy []networkingv1.NetworkPolicyPeer) {
+	for _, egress := range app.Spec.AccessPolicy.Egress.Rules {
+		networkPolicyPeer := networkingv1.NetworkPolicyPeer{
 			PodSelector: &metav1.LabelSelector{
 				MatchLabels: map[string]string{
-					"app": ingress.Application,
+					"app": egress.Application,
 				},
 			},
 		}
 
-		if ingress.Namespace != "" {
+		if egress.Namespace != "" {
 			networkPolicyPeer.NamespaceSelector = &metav1.LabelSelector{
 				MatchLabels: map[string]string{
-					"name": ingress.Namespace,
+					"name": egress.Namespace,
 				},
 			}
 		}
 
-		networkPolicy.Spec.Ingress = append(networkPolicy.Spec.Ingress, networking.NetworkPolicyIngressRule{
-			From: []networking.NetworkPolicyPeer{
-				networkPolicyPeer,
-			},
-		})
+		networkPolicy = append(networkPolicy, networkPolicyPeer)
 	}
+
+	return
 }
 
-// allowAll removes a given PolicyType from an NetworkPolicy, which again
-// opens up for all traffic of that policy type
-func allowAll(allowedPolicyType networking.PolicyType, networkPolicy *networking.NetworkPolicy) {
-	policies := networkPolicy.Spec.PolicyTypes
-	for index, policyType := range policies {
-		if policyType == allowedPolicyType {
-			policies[index] = policies[len(policies)-1]
-			networkPolicy.Spec.PolicyTypes = policies[:len(policies)-1]
-			return
+func egressPolicy(app *nais.Application) *[]networkingv1.NetworkPolicyEgressRule {
+	if app.Spec.AccessPolicy.Egress.AllowAll {
+		return &[]networkingv1.NetworkPolicyEgressRule{{}} // æh funker dette?
+	}
+
+	if len(app.Spec.AccessPolicy.Egress.Rules) > 0 {
+		policies := networkingv1.NetworkPolicyEgressRule{
+			To: addNetworkPolicyEgressRules(app),
 		}
+		return &[]networkingv1.NetworkPolicyEgressRule{policies}
 	}
+
+	return &[]networkingv1.NetworkPolicyEgressRule{}
 }
 
-func NetworkPolicy(app *nais.Application) (networkPolicy *networking.NetworkPolicy) {
-	networkPolicy = getDefaultNetworkPolicy(app)
-	if len(app.Spec.AccessPolicy.Ingress.Rules) > 0 {
-		addNetworkPolicyIngressRules(networkPolicy, app)
-	}
-	if len(app.Spec.AccessPolicy.Egress.Rules) > 0 || app.Spec.AccessPolicy.Egress.AllowAll {
-		allowAll(networking.PolicyTypeEgress, networkPolicy)
+func networkPolicySpec(app *nais.Application) *networkingv1.NetworkPolicySpec {
+	ingressPolicy := ingressPolicy(app)
+	egressPolicy := egressPolicy(app)
```

Comment:
Disse to kan flyttes rett inn i neste objekt, siden vi ikke gjør noen endringer.